packages/analytics/modules/analytics/client.ts

```diff
@@ -13,6 +13,40 @@ import { browser } from '@wxt-dev/browser';
 
 const ANALYTICS_PORT = '@wxt-dev/analytics';
 
+// Based on https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/manifest.json/background#browser_support
+function isBackgroundScript(): boolean {
+  if (!browser?.runtime?.getManifest || !location.protocol.endsWith('extension:')) {
+    return false;
+  }
+
+  const { background, background_page: backgroundPageWXT, manifest_version } = browser.runtime.getManifest();
+  // Misonfiguration or Non-WXT Contenxt Check
+  // Force check page attribute for possible configurations
+  const backgroundPageManifest = (background as any)?.page;
+
+  const backgoundPageorScriptsCheck =
+    location.pathname === '/_generated_background_page.html' ||
+    (backgroundPageManifest && location.pathname.endsWith(backgroundPageManifest)) ||
+    (backgroundPageWXT && location.pathname.endsWith(backgroundPageWXT));
```

Comment:
I need help over here. I couldn't find `background_page` property on the manifest anywhere, so I'm not sure if this internal and needs to be checked

Reply:
Not sure where you found this property in the first place... Isn't it `background.page`?

Reply:
Yes, you are right. I earlier thought that having `background_page` set in the manifest would generate `_generated_background_page.html` but it's  actually when you only specify '.js' files in the scripts